keras_cv/layers/preprocessing/center_crop.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+from tensorflow import keras
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class CenterCrop(BaseImageAugmentationLayer):
+    """A preprocessing layer which crops images.
+
```

Comment:
This layers crops the central portion of the images to a target size. If an image is smaller than the target size, it will be resized and cropped so as to return the largest possible window in the image that matches the target aspect ratio.

Reply:
>  it will be resized..

Do we need to introduce some safety warning log about the ratio? As It could happen to have extreme values processing bulk datasets...

Reply:
...Or at least a explicit flag to enable these cases..

Reply:
Is it possible to have more details? Should be raise a warning when the aspect ratio is too much different or when the size is too different?